Marlin/Marlin_main.cpp

```diff
@@ -3205,25 +3207,32 @@ static void homeaxis(const AxisEnum axis) {
       current_position[E_AXIS] += (swapping ? swap_retract_length : retract_length) / volumetric_multiplier[active_extruder];
       sync_plan_position_e();
       prepare_move_to_destination();
-
-      // Is a Z hop set, and has the hop not yet been done?
-      if (has_zhop) {
```

Comment:
The machine is the machine 
The gcode is the gcode
If the machine make a hop , than the machine can recover and it's all
If gcode files have no respect for the machine , it's not a marlin problem